### PR TITLE
Add ConfigMap docs

### PR DESCRIPTION
**Pull request by pmorie**
@adellape

### CODE REVIEW

**Comment by adellape** [comment]
@pmorie For OSE, this should target 3.2, yes? Looking at https://trello.com/c/K16g7VGi.

File: dev_guide/configmaps.adoc [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+== Overview
+
+The `ConfigMap` object type provides a mechanism to hold non-sensitive
+configuration information such as whole configuration files, property values,
+etc. ConfigMaps separate configuration data from the pods that use it and can
+be mounted into containers using a volume plug-in or injected as an
+environment variable.  This topic discusses important properties of ConfigMaps
+and provides an overview on how developers can use them.
+
+ConfigMaps is similar to the link:secrets.html[Secrets] type, but is
```
Comment (adellape) [inline dev_guide/configmaps.adoc]: s/is/are/

x2

File: dev_guide/configmaps.adoc [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+  "metadata": {
+    "name": "example-configmap"
+  },
+  "namespace": "my-namespace",
+  "data": { <1>
+    "should-work-right": "true",
+    "sometimes-be-confusing": "false"
+    "whole-files-supported": "supported=why yes they are\nhow-enthusiastically=gladly"
+    }
+}
```
Comment (adellape) [inline dev_guide/configmaps.adoc]: Show YAML instead.

File: dev_guide/configmaps.adoc [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+----
+<1> The allowable format for the keys in the `data` field must meet the
+guidelines in the *DNS_SUBDOMAIN* value in
+https://github.com/GoogleCloudPlatform/kubernetes/blob/v1.0.0/docs/design/identifiers.md[the
```
Comment (adellape) [inline dev_guide/configmaps.adoc]: Is v1.0.0 intentional here or just an old link? I see this callout came from the Secrets topic.

File: dev_guide/configmaps.adoc [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+[[configmaps-in-environment-variables]]
+
+=== ConfigMaps in Environment Variables
+
+ConfigMaps can also be consumed in environment variables.  Environment variables populated by ConfigMaps
+are not updated when the ConfigMap is updated.
+
+See link:#examples[Examples].
+
+[[restrictions]]
```
Comment (adellape) [inline dev_guide/configmaps.adoc]: Give this a more unique ID, like `configmap-restrictions`.

File: dev_guide/configmaps.adoc [hunk 5]
```diff
@@ -0,0 +1,285 @@
...
+Individual ConfigMaps are limited to 1MB in size. This is to discourage the
+creation of large ConfigMaps that would exhaust apiserver and kubelet memory.
+However, creation of a number of smaller ConfigMaps could also exhaust memory.
+
+[[configmap-data-keys]]
+
+=== ConfigMap Data Keys
+ConfigMap keys must be in a DNS subdomain.
+
+[[examples]]
```
Comment (adellape) [inline dev_guide/configmaps.adoc]: More unique ID (`configmap-examples`), and update the related xref above.

**Comment by adellape** [comment]
@pmorie See inline. I didn't go all out w/ the red pen since you mentioned that you would be following up with overhaul edits to this and the Secret topics together. After this and your follow-up are merged, a docs person will continue from there.

File: dev_guide/configmaps.adoc (hunk 3 above)
Comment (pmorie) [inline dev_guide/configmaps.adoc]: old link, copied from secrets (it's still that way in the secrets docs -- how about i fix them together after this merges.

File: dev_guide/configmaps.adoc (hunk 4 above)
Comment (pmorie) [inline dev_guide/configmaps.adoc]: 10-4

File: dev_guide/configmaps.adoc (hunk 3 above)
Comment (adellape) [inline dev_guide/configmaps.adoc]: :+1:

**Comment by pmorie** [comment]
@adellape I think after our earlier convo that I should do the doc overhaul in this PR.

**Comment by pmorie** [comment]
@adellape PTAL

File: dev_guide/configmaps.adoc [hunk 6]
```diff
@@ -0,0 +1,616 @@
...
+Many applications require configuration via some combination of config files, command line
+arguments, and environment variables.  These configuration artifacts should be decoupled from image
+content in order to keep containerized applications portable.  The ConfigMap API resource provides
+mechanisms to inject containers with configuration data while keeping containers agnostic of
+OpenShift.  ConfigMap can be used to store fine-grained information like individual properties or
+coarse-grained information like entire config files or JSON blobs.
+
+The ConfigMap API resource holds key-value pairs of configuration data that can be consumed in pods
+or used to store configuration data for system components such as controllers.  ConfigMap is similar
+to link:Secrets#secrets), but designed to more conveniently support working with strings that do not
```
Comment (adellape) [inline dev_guide/configmaps.adoc]:
```
similar to link:../dev_guide/secrets.html[secrets], but
```

File: dev_guide/configmaps.adoc [hunk 7]
```diff
@@ -0,0 +1,616 @@
...
+
+1.  Populate the value of environment variables
+2.  Set command-line arguments in a container
+3.  Populate config files in a volume
+
+Both users and system components may store configuration data in ConfigMap.
+
+[[creating-configmaps]]
+
+## Creating ConfigMaps
```
Comment (adellape) [inline dev_guide/configmaps.adoc]: Overall comment throughout, use `=`s instead of `#`s for section headers.

File: dev_guide/configmaps.adoc [hunk 8]
```diff
@@ -0,0 +1,616 @@
...
+
+You can use the `kubectl create configmap` command to create configmaps easily from literal values,
```
Comment (adellape) [inline dev_guide/configmaps.adoc]: Looks like this can be `oc create configmap` now?

Comment (adellape) [inline dev_guide/configmaps.adoc]: Seems to be a few other references to `kubectl` remaining throughout.

File: dev_guide/configmaps.adoc [hunk 9]
```diff
@@ -0,0 +1,616 @@
...
+how.nice.to.look=fairlyNice
+----
+====
+
+The `oc create configmap` command can be used to create a ConfigMap holding the content of each
+file in this directory:
+
+====
+----
+$ oc create configmap game-config --from-file=examples/
```
Comment (adellape) [inline dev_guide/configmaps.adoc]: example-files/

File: dev_guide/configmaps.adoc [hunk 10]
```diff
@@ -0,0 +1,616 @@
...
+====
+----
+$ oc describe configmaps game-config
+Name:           game-config
+Namespace:      default
+Labels:         <none>
+Annotations:    <none>
+
+Data
+====
```
Comment (adellape) [inline dev_guide/configmaps.adoc]: This legitimate `====` in the configmap output is messing up the example/code blocks. Haven't been able to figure out yet a way to escape this. Maybe the easiest solution for now would be to remove the example block (`====`) around it and just let this output only be in a code block (`----`). Seems to render fine that way, at least with asciibinder (and hopefully on the Customer Portal).

File: dev_guide/configmaps.adoc [hunk 11]
```diff
@@ -0,0 +1,616 @@
...
+game.properties:        121 bytes
+ui.properties:          83 bytes
+----
+====
+
+You can see the two keys in the map are created from the filenames in the directory we pointed
+kubectl to.  Since the content of those keys may be large, in the output of `kubectl describe`,
+you'll see only the names of the keys and their sizes.
```
Comment (adellape) [inline dev_guide/configmaps.adoc]: s/kubectl/oc/g

File: dev_guide/configmaps.adoc [hunk 12]
```diff
@@ -0,0 +1,616 @@
...
+        "selfLink": "/api/v1/namespaces/default/configmaps/example-redis-config",
+        "uid": "07fd0419-d97b-11e5-b443-68f728db1985",
+        "resourceVersion": "15",
+        "creationTimestamp": "2016-02-22T15:43:34Z"
+    },
+    "data": {
+        "redis-config": "maxmemory 2mb\nmaxmemory-policy allkeys-lru\n"
+    }
+}
+----
```
Comment (adellape) [inline dev_guide/configmaps.adoc]: YAML?

File: dev_guide/configmaps.adoc [hunk 13]
```diff
@@ -0,0 +1,616 @@
...
+====
+
+## Restrictions
+
+ConfigMaps must be created before they are consumed in pods.  Controllers may be written to tolerate
+missing configuration data; consult individual components configured via ConfigMap on a case-by-case
+basis.
+
+ConfigMaps reside in a namespace.   They can only be referenced by pods in the same namespace.
```
Comment (adellape) [inline dev_guide/configmaps.adoc]: Can we s/namespace/project/g here? Trying to err on the side of referring to projects when we can, but mention "namespace" when we can't avoid it (i.e., `--all-namespaces`).

File: dev_guide/configmaps.adoc [hunk 14]
```diff
@@ -0,0 +1,616 @@
...
+
+Kubelet only supports use of ConfigMap for pods it gets from the API server.  This includes any pods
```
Comment (adellape) [inline dev_guide/configmaps.adoc]: The Kubelet?

**Comment by adellape** [comment]
@pmorie Some comments inline.

**Comment by pmorie** [comment]
@adellape Thanks for the comments and sorry for the latency; will get these addressed ASAP.

**Comment by pmorie** [comment]
@adellape feedback addressed, PTAL

**Comment by adellape** [comment]
@pmorie Went ahead and pulled this into https://github.com/openshift/openshift-docs/pull/1896 to put my follow-up edits in at the same time. PTAL there, closing here.

**Pull request closed without merging**